google/auth/pluggable.py

```diff
@@ -212,18 +212,15 @@ def retrieve_subject_token(self, request):
         env = os.environ.copy()
         env["GOOGLE_EXTERNAL_ACCOUNT_AUDIENCE"] = self._audience
         env["GOOGLE_EXTERNAL_ACCOUNT_TOKEN_TYPE"] = self._subject_token_type
-        env["GOOGLE_EXTERNAL_ACCOUNT_ID"] = self.external_account_id
+        env["GOOGLE_EXTERNAL_ACCOUNT_ID"] = self.external_account_id or ""
         env["GOOGLE_EXTERNAL_ACCOUNT_INTERACTIVE"] = "1" if self.interactive else "0"
         env["GOOGLE_EXTERNAL_ACCOUNT_REVOKE"] = 0
-
-        if self._service_account_impersonation_url is not None:
-            env[
-                "GOOGLE_EXTERNAL_ACCOUNT_IMPERSONATED_EMAIL"
-            ] = self.service_account_email
-        if self._credential_source_executable_output_file is not None:
-            env[
-                "GOOGLE_EXTERNAL_ACCOUNT_OUTPUT_FILE"
-            ] = self._credential_source_executable_output_file
+        env["GOOGLE_EXTERNAL_ACCOUNT_IMPERSONATED_EMAIL"] = (
+            self.service_account_email or ""
```

Comment:
why was this switched to filling with "" vs just not setting it at all?

Reply:
Redundant change by an unsophisticated idea. I'll revert.